### PR TITLE
Add sonarcloud workflow

### LINKED ISSUE #595: Configure CodeQL and sonarcloud
following the outline in the suggestion to add those tools to the copier template - it should be tested first 
https://github.com/DiamondLightSource/python-copier-template/issues/180

a link to the current report could be provided in the README

### PR DESCRIPTION
**Pull request by stan-dot**
Fixes #595 (fully this time)

### CODE REVIEW

**Comment by stan-dot** [comment]
@DiamondJoseph *I* do not have one, plausibly @callumforrester  signed up for the account for the DLS org and the key can be found / generated there

**Comment by callumforrester** [comment]
I did not...
You will have to figure out who owns it and approach them

**Comment by stan-dot** [comment]
need to add org key

**Comment by stan-dot** [comment]
now debugging this
> **08:59:50.762 ERROR main component does not belong to specified organization**


which surprisingly only pops up after analysis has run succesfully for 40 seconds

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+on:
+  # Trigger analysis when pushing in master or pull requests, and when creating
+  # a pull request.
+  push:
+    branches:
+      - master
```
Comment (callumforrester) [inline .github/workflows/sonarcloud.yml:6]:
```suggestion
      - main
```

Comment (stan-dot) [reply .github/workflows/sonarcloud.yml]: done

**Review by keithralphs — APPROVED** [review]

**Pull request merged**